MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -0,0 +1,254 @@
+#pragma warning disable CA1062 // Validate arguments of public methods
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Runtime.CompilerServices;
+using NetCollection = System.Collections.ObjectModel.Collection<MediaBrowser.Common.Net.IPObject>;
+
+namespace MediaBrowser.Common.Net
+{
+    /// <summary>
+    /// Defines the <see cref="NetworkExtensions" />.
+    /// </summary>
+    public static class NetworkExtensions
+    {
+        /// <summary>
+        /// Add an address to the collection.
+        /// </summary>
+        /// <param name="source">The <see cref="NetCollection"/>.</param>
+        /// <param name="ip">Item to add.</param>
+        public static void AddItem(this NetCollection source, IPAddress ip)
+        {
+            if (!source.ContainsAddress(ip))
+            {
+                source.Add(new IPNetAddress(ip, 32));
+            }
+        }
+
+        /// <summary>
+        /// Add multiple items to the collection.
+        /// </summary>
+        /// <param name="destination">The <see cref="NetCollection"/>.</param>
+        /// <param name="source">Item to add.</param>
+        /// <returns>Return the collection.</returns>
+        public static NetCollection AddRange(this NetCollection destination, IEnumerable<IPObject> source)
+        {
+            foreach (var item in source)
+            {
+                destination.Add(item);
+            }
+
+            return destination;
+        }
+
+        /// <summary>
+        /// Adds a network to the collection.
+        /// </summary>
+        /// <param name="source">The <see cref="NetCollection"/>.</param>
+        /// <param name="item">Item to add.</param>
+        public static void AddItem(this NetCollection source, IPObject item)
+        {
+            if (!source.ContainsAddress(item))
+            {
+                source.Add(item);
+            }
+        }
+
+        /// <summary>
+        /// Converts this object to a string.
+        /// </summary>
+        /// <param name="source">The <see cref="NetCollection"/>.</param>
+        /// <returns>Returns a string representation of this object.</returns>
+        public static string Readable(this NetCollection source)
+        {
+            string output = "[";
```

Comment:
Use `StringBuilder` when concatenating many strings

Reply:
For the number of iterations, a SB is way overkill. (You're probably only talking about 5 iterations).